stratum/src/lib.rs

```diff
@@ -83,48 +120,83 @@ pub struct Stratum {
 	workers: Arc<RwLock<HashMap<SocketAddr, String>>>,
 	/// Secret if any
 	secret: Option<H256>,
+	/// Dispatch notify couinter
+	notify_counter: RwLock<u32>,
+	/// Message dispatcher (tcp/ip service)
+	tcp_dispatcher: Dispatcher,
 }
 
+const NOTIFY_CONTER_INITIAL: u32 = 16;
+
 impl Stratum {
 	pub fn start(
 		addr: &SocketAddr,
 		dispatcher: Arc<JobDispatcher>,
 		secret: Option<H256>,
-	) -> Result<Arc<Stratum>, jsonrpc_tcp_server::Error> {
+	) -> Result<Arc<Stratum>, Error> {
+
 		let rpc = Arc::new(StratumRpc {
 			stratum: RwLock::new(None),
 		});
-		let mut delegate = IoDelegate::<StratumRpc>::new(rpc.clone());
-		delegate.add_method("miner.subscribe", StratumRpc::subscribe);
-		delegate.add_method("miner.authorize", StratumRpc::authorize);
-
-		let mut handler = IoHandler::default();
+		let mut delegate = IoDelegate::<StratumRpc, SocketMetadata>::new(rpc.clone());
+		delegate.add_method_with_meta("mining.subscribe", StratumRpc::subscribe);
+		delegate.add_method_with_meta("mining.authorize", StratumRpc::authorize);
+		delegate.add_method_with_meta("mining.submit", StratumRpc::submit);
+		let mut handler = MetaIoHandler::<SocketMetadata>::with_compatibility(Compatibility::V1);
```

Comment:
It just does not care about any protocol notes, so it is V1
But V2 is also fine, so i will add for clarity, thanx